# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/ash/src/vk/platform_types.rs
# Commit: ef5dc3e04e5f
# Full Hash: ef5dc3e04e5f271eea0636ab3a495e95cc912f1d
# Author: Dzmitry Malyshau <dmalyshau@mozilla.com>
# Date: 2021-09-04 09:40:24
# Regressor Bug: 1726626
# File Overlap Count: 3
# Description:
#   Bug 1726626 - Move gfx/wgpu into a 3rd party dependency r=jgilbert,bholley
#   
#   This update makes wgpu a vendored dependency instead of having it in gfx/wgpu.
#   
#   ## Notes
# ==============================================================================

diff --git a/third_party/rust/ash/src/vk/platform_types.rs b/third_party/rust/ash/src/vk/platform_types.rs
--- a/third_party/rust/ash/src/vk/platform_types.rs
+++ b/third_party/rust/ash/src/vk/platform_types.rs
@@ -19,16 +19,23 @@ pub type wl_display = c_void;
 pub type wl_surface = c_void;
 pub type HANDLE = *mut c_void;
 pub type HMONITOR = HANDLE;
 pub type DWORD = c_ulong;
 pub type LPCWSTR = *const u16;
 #[allow(non_camel_case_types)]
 pub type zx_handle_t = u32;
 #[allow(non_camel_case_types)]
-pub type SECURITY_ATTRIBUTES = ();
+pub type _screen_context = c_void;
+#[allow(non_camel_case_types)]
+pub type _screen_window = c_void;
+#[allow(non_camel_case_types)]
+pub type SECURITY_ATTRIBUTES = c_void;
+// Opaque types
 pub type ANativeWindow = c_void;
 pub type AHardwareBuffer = c_void;
 pub type CAMetalLayer = c_void;
+// This definition is behind an NDA with a best effort guess from
+// https://github.com/google/gapid/commit/22aafebec4638c6aaa77667096bca30f6e842d95#diff-ab3ab4a7d89b4fc8a344ff4e9332865f268ea1669ee379c1b516a954ecc2e7a6R20-R21
 pub type GgpStreamDescriptor = u32;
 pub type GgpFrameToken = u64;
 pub type IDirectFB = c_void;
 pub type IDirectFBSurface = c_void;